src/Swashbuckle.AspNetCore.SwaggerGen/DependencyInjection/SwaggerGenOptions.cs

```diff
@@ -32,5 +32,11 @@ public class FilterDescriptor
         public object[] Arguments { get; set; }
 
         public object FilterInstance { get; set; }
+
+        internal bool IsAssignableTo(Type type)
+        {
+            return FilterInstance != null && type.IsInstanceOfType(FilterInstance) ||
+                   Type != null && Type.IsAssignableTo(type);
```

Comment:
Add parenthesis to make the order of operations obvious:
```suggestion
            return (FilterInstance != null && type.IsInstanceOfType(FilterInstance)) ||
                   (Type != null && Type.IsAssignableTo(type));
```